core/src/main/java/io/onetable/iceberg/IcebergClient.java

```diff
@@ -82,8 +84,18 @@ public IcebergClient() {}
       IcebergPartitionSpecSync partitionSpecSync,
       IcebergDataFileUpdatesSync dataFileUpdatesExtractor,
       IcebergTableManager tableManager) {
+    this.tableName = perTableConfig.getTableName();
```

Comment:
Why move this out of the `_init` method?

Reply:
@the-other-tim-brown - not sure I am answering the right question but... because PerTableConfig was removed from the init() which calls _init() and the ctor's also call _init so they should handle the PerTableConfig specifics. This is assuming that the ctor's are mostly (if not entirely) used by tests and they would be setting these things there rather than via the factory.

Maybe you see something wrong with what I did there that I am still missing?